crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt

```diff
@@ -0,0 +1,10 @@
+package net.corda.v5.crypto.merkle
+
+import net.corda.v5.crypto.SecureHash
+
+interface MerkleTree {
+    val leaves: List<ByteArray>
+    val digestProvider: MerkleTreeHashDigestProvider
```

Comment:
You are making the digest provider as part of the tree structure and mixing it with the properties related to the tree itself